Tests/test_file_gif.py

```diff
@@ -507,6 +505,35 @@ def test_save_I(self):
         reloaded = Image.open(out)
         self.assert_image_equal(reloaded.convert('L'), im.convert('L'))
 
+    def test_getdata(self):
+        # test getheader/getdata against legacy values
+        # Create a 'P' image with holes in the palette
+        im = Image._wedge().resize((16,16))
+        im.putpalette(ImagePalette.ImagePalette('RGB'))
+        im.info = {'background':0}
+        
+        passed_palette = bytes(bytearray([255-i//3 for i in range(768)]))
+
+        GifImagePlugin._FORCE_OPTIMIZE = True
+        try:
+            h = GifImagePlugin.getheader(im, passed_palette)
+            d = GifImagePlugin.getdata(im)
+
+            import cPickle as pickle
+            # Enable to get target values on pre-refactor version
+            # with open('Tests/images/gif_header_data.pkl', 'wb') as f:
+            #    f.write(pickle.dumps((h,d)))
+            with open('Tests/images/gif_header_data.pkl', 'rb') as f:
+                (h_target,d_target) = pickle.loads(f.read())
```

Comment:
Yeah, but it's not really worth that much bother. `pickle` is in both, it's clear, it works, and cPickle is only going to save milliseconds on py2.7 test runs.